AttitudeManager/FlightModes/Src/fbwa.cpp

```diff
@@ -0,0 +1,37 @@
+#include "fbwa.hpp"
+#include "CommonDataTypes.hpp"
+#include "AM.hpp"
+#include <cassert>
+
+namespace AM {
+    AttitudeManagerInput FBWA::run(const AttitudeManagerInput& input) {
+        
+        AttitudeManagerInput mappedOutputs;
+
+        // Bounded checking to make sure input is within max and min
+        assert(input.pitch <= AM::AttitudeManager::INPUT_MAX && input.pitch >= AM::AttitudeManager::INPUT_MIN);
+        assert(input.roll <= AM::AttitudeManager::INPUT_MAX && input.roll >= AM::AttitudeManager::INPUT_MIN);
+        assert(input.yaw <= AM::AttitudeManager::INPUT_MAX && input.yaw >= AM::AttitudeManager::INPUT_MIN);
+        assert(input.throttle <= AM::AttitudeManager::INPUT_MAX && input.throttle >= AM::AttitudeManager::INPUT_MIN);
+
+        // Mapping inputs to outputs with ratios of min-max
+        // Mapping maintains 0->0 I/O
+        mappedOutputs.pitch = input.pitch < 0 ? input.pitch * (fbwa_control_limits.pitchLimit.min /  AM::AttitudeManager::INPUT_MIN) : input.pitch * (fbwa_control_limits.pitchLimit.max /  AM::AttitudeManager::INPUT_MAX);        
+        mappedOutputs.roll = input.roll < 0 ? input.roll * (fbwa_control_limits.rollLimit.min /  AM::AttitudeManager::INPUT_MIN) : input.roll * (fbwa_control_limits.rollLimit.max /  AM::AttitudeManager::INPUT_MAX);        
+        mappedOutputs.yaw = input.yaw < 0 ? input.yaw * (fbwa_control_limits.yawLimit.min /  AM::AttitudeManager::INPUT_MIN) : input.yaw * (fbwa_control_limits.rollLimit.max /  AM::AttitudeManager::INPUT_MAX);
```

Comment:
Typo in yaw calculation. `fbwa_control_limits.rollLimit.max` should be `fbwa_control_limits.yawLimit.max`